### PR TITLE
Wire In SRAT Catchment API

### PR DESCRIPTION
**Pull request by arottersman**
## Overview

In sub-basin modeling mode, we run the GWLF-E model on each of a HUC-8/HUC-10's composite HUC-12s. The API created at Drexel the "SRAT Catchment API" calculates the attenuates model results for each HUC-12 and its contained NHD Plus catchments.

This PR calls the API with an AOI's HUC-12 results. The API results return to the browser but are not formatted or displayed. In later work we will sum the per HUC-12 attenuated results to get results for the total AOI.

Connects #2612 

### Demo

screen shot 2018-02-07 at 11 47 10 am

[sample_subbasin_response.json.txt](https://github.com/WikiWatershed/model-my-watershed/files/1704409/sample_subbasin_response.json.txt)

### Notes

- [ ] I will add the API key to our fileshare's `default.yml` when this PR has been approved.

## Testing Instructions

 * Pull this branch and provision your app and worker VMs with the api key on Last Pass under "SRAT Catchment API (Staging)" (the worker should be reprovisioned for the security package installation — you can also install these manually with `pip` if you like)
       ```bash
       > MMW_SRAT_CATCHMENT_API_KEY=*** vagrant reload app worker --provision
       ```

<details>
<summary> <strong>NB:</strong> You may wish to give your worker VM two CPUs and run these steps with two celery threads to speed up sub-basin modeling results. This process will involve ...</summary>

```diff
diff --git a/Vagrantfile b/Vagrantfile
index 008818c..1c9dcd4 100644
--- a/Vagrantfile
+++ b/Vagrantfile
@@ -75,6 +75,7 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |conf
     services.vm.provider "virtualbox" do |v|
       v.memory = 1024
+      v.cpus = 2
     end

     services.vm.provision "ansible" do |ansible|
```

Running `./scripts/debugcelery.sh` and then adding an additional thread:

```sh
> vagrant ssh worker -c \
    "cd /opt/app/ && envdir /etc/mmw.d/env celery -A "mmw.celery:app" worker -l debug -n debug2@%n"
```

</details>
</br>

* Turn the subbasin feature flag on

    ```bash
    > ./scripts/toggle_feature.sh subbasin
    ```
* In the app with your browser's dev tools open, select a HUC-10 and run the MapShed model
* Nothing will be displayed in the sidebar, but the requests should complete and you should see a `results.huc12` object.

### CODE REVIEW

File: deployment/cfn/application.py [hunk 1]
```diff
@@ -608,7 +620,14 @@ def get_cloud_config(self, tile_distribution_endpoint):
                 '  - path: /etc/mmw.d/env/MMW_HYDROSHARE_SECRET_KEY\n',
                 '    permissions: 0750\n',
                 '    owner: root:mmw\n',
-                '    content: ', Ref(self.hydroshare_secret_key), '\n',
```
Comment (rajadain) [inline deployment/cfn/application.py]: Was this line accidentally deleted?

Comment (arottersman) [reply deployment/cfn/application.py]: No, it was intentional sabotage, but looks like I've been caught. Will add a fixup.

**Comment by rajadain** [comment]
Reprovisioned `app` and `worker` with the SRAT key set, ran two parallel Celery instances, see this in the output as a job fails:

```
[2018-02-07 16:56:08,726: ERROR/ForkPoolWorker-2] Request to SRAT Catchment API failed: [Errno 1] _ssl.c:510: error:14077410:SSL routines:SSL23_GET_SERVER_HELLO:sslv3 alert handshake failure
```

**Comment by arottersman** [comment]
Okay, I think I've got it.

Fixup specifies `requests[security]` add ons which should use the correct version of `pyOpenSSl`. I was unable to load the app due to a fiona error. I had to upgrade `six` to get it to work, so that is also included in the commit.

There's an additional fixup that throughs an exception from the srat task if the API call returns with a status code that's not 200.

**Comment by rajadain** [comment]
Trying it out again.

File: src/mmw/apps/modeling/tasks.py [hunk 2]
```diff
@@ -265,6 +265,9 @@ def run_srat(watersheds):
         result = r.json()
     except ValueError:
         logger.error('SRAT Catchment API did not return JSON')
+    
+    if (r.status_code != 200):
+        raise Exception('SRAT Catchment API request failed: %s' % result)
```
Comment (rajadain) [inline src/mmw/apps/modeling/tasks.py]: Adding these lines makes SRAT Celery task never finish. Commenting these out makes it work again. Would recommend just logging it for now.

File: deployment/default.yml.example [hunk 3]
```diff
@@ -65,5 +65,7 @@ ITSIBaseURL: ''
 ITSISecretKey: ''
 HydroShareBaseURL: ''
 HydroShareSecretKey: ''
+SRATCatchmentAPIURL: ''
```
Comment (hectcastro) [inline deployment/default.yml.example]: Default values for these two need to be in `/var/lib/jenkins/.aws/staging.yml` on CivicCI for staging deployments to work again.

Comment (rajadain) [reply deployment/default.yml.example]: Good catch. Added the values and triggered a deploy.

Comment (arottersman) [reply deployment/default.yml.example]: Thanks!!

**Pull request merged**